src/client/client-wrapper.ts

```diff
@@ -87,4 +92,22 @@ export class ClientWrapper {
 
     return result;
   }
+
+  public async evaluateUrls(urls: string[]) {
+    const brokenUrls = [];
+
+    return new Promise((resolve) => {
+      Bluebird.each(urls, (url) => {
+        return this.request.get(url).then((res) => {
+        }).catch(() => {
+          brokenUrls.push(url);
+        });
+      }).then(() => {
+        resolve(brokenUrls);
+      }).catch(() => {
+        resolve(brokenUrls);
```

Comment:
Mmm!  I think you could achieve full concurrency, while retrieving all broken URLs, and also get rid of the Bluebird dependency using something like what you've done here, but going back to `Promise.all`:

Something like: 
```javascript
public async evaluateUrls(urls: string[]) {
  const brokenUrls = [];

  await Promise.all(urls.map((url) => {
    return new Promise((resolve) => {
      this.request.get(url).then(resolve).catch(() => {
        brokenUrls.push(url);
        resolve();
      });
    });
  });

  return brokenUrls.length > 0 ? Promise.reject(brokenUrls) : Promise.resolve();
}
```

Reply:
Oh did you do this already?

Reply:
I did not--but you could rebase your fork and open another PR